xarray/core/groupby.py

```diff
@@ -876,6 +888,166 @@ def reduce_array(ar):
 
         return self.map(reduce_array, shortcut=shortcut)
 
+    if XARRAY_NUMPY_GROUPIES:
+
+        def count(self, dim=None, *, keep_attrs=True, fill_value=None):
```

Comment:
In a future version it could be nice to have docstrings. One options is generating these like we do `_typed_ops.py`, a bit like a macro. I find that works better than our "injections" because static tools can still find the docstrings & types.